### ISSUE TITLE
Know when tvOSFocusGuide gets focused

### ISSUE THREAD

**Issue by ericlifs** [react-native-tvos/react-native-tvos#305]
### Description

Hi everyone, thanks for this awesome tool. 

I have a question regarding the tvOSFocusGuide component and a feature I have to build, which I'm not certain whether it's feasible or not.

I have a layout similar to this one: 
![Simulator Screen Shot - Apple TV - 2022-04-28 at 16 50 45](https://user-images.githubusercontent.com/7376892/165836058-0c15af83-d931-4352-b5d7-78ecf53ddc31.png)

The red box is the tvOSFocusGuide component I have added in order to jump from any button from zone 3 to the zone 1. That's working perfectly ✅ 

The thing is that the zone 1 can be hidden like this:
![Simulator Screen Shot - Apple TV - 2022-04-28 at 16 50 50](https://user-images.githubusercontent.com/7376892/165836344-6d9d1586-b2c3-4349-88b8-db2ef4a1d96b.png)

And in that particular case, before jumping to zone 1 I first need to show it.

Here's where my question comes: is there a way of knowing when the tvOSFocusGuide component gets focused in order to trigger that visibility change? If not, is there any recommended workaround to do something similar to this behavior?

Thank you very much,

Eric.

### Version

-

### Output of `npx react-native info`

-

### Steps to reproduce

-

**Comment by douglowder** [comment]
Interesting problem...

In fact the focus guide component never actually receives focus itself, it just hands off the focus to its destination. The destination has to actually exist and be rendered in order to be a valid destination. Also, there is no way to know when the focus guide is triggered. I'll have to think about this a bit.

**Comment by douglowder** [comment]
@ericlifs my first idea for you to try, is to add `useTVEventHandler()` and, if zone 1 is hidden and you see an event with `eventType === 'left'`, then show zone 1 so the focus guide will work.

**Comment by ericlifs** [comment]
@douglowder yes, that was my initial thought as well, but the thing is that I have to differentiate which is the current active "zone", because a `left` event while the zone 2 being active may mean that you are jumping from a zone 2 button to another, so I also have to know if I'm currently focusing a zone 3 button and a `left` event is triggered. And that's where things start to get complicated (I'd simplified the layout) and sounds more like a hacky workaround than a real stable solution.

I might end up doing that because that's the only way to do it, but I would rather take a nicer and tidier approach if possible.

**Comment by douglowder** [comment]
That could be done with the `onFocus()` and `onBlur()` handlers in your buttons updating a state, but yes there would be a little complexity involved. I'll think on this some more :)
